examples/vulkan/vk_hotload.rs

```diff
@@ -0,0 +1,246 @@
+extern crate nannou;
+extern crate shade_runner;
+
+use nannou::prelude::*;
+use std::cell::RefCell;
+use std::ffi::CStr;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+struct Model {
+    render_pass: Arc<vk::RenderPassAbstract + Send + Sync>,
+    pipeline: Option<Arc<vk::GraphicsPipelineAbstract + Send + Sync>>,
+    vertex_buffer: Arc<vk::CpuAccessibleBuffer<[Vertex]>>,
+    view_fbo: RefCell<ViewFbo>,
+    shade_watcher: shade_runner::Watch,
+    shade_msg: shade_runner::Message,
+    vert_shader: Arc<vk::pipeline::shader::ShaderModule>,
+    frag_shader: Arc<vk::pipeline::shader::ShaderModule>,
+    device: Arc<vk::Device>,
+}
+
+#[derive(Debug, Clone)]
+struct Vertex {
+    position: [f32; 2],
+}
+
+vk::impl_vertex!(Vertex, position);
+
+fn main() {
+    nannou::app(model).view(view).update(update).run();
+}
+
+fn model(app: &App) -> Model {
+    app.new_window()
+        .with_dimensions(512, 512)
+        .view(view)
+        .build()
+        .unwrap();
+
+    // The gpu device associated with the window's swapchain
+    let device = app.main_window().swapchain().device().clone();
+
+    // We now create a buffer that will store the shape of our triangle.
+    let vertex_buffer = {
+        let positions = [[-0.5, -0.25], [0.0, 0.5], [0.25, -0.1]];
+        let vertices = positions.iter().map(|&position| Vertex { position });
+        vk::CpuAccessibleBuffer::from_iter(device.clone(), vk::BufferUsage::all(), vertices)
+            .unwrap()
+    };
+    let nannou_root = std::env::current_dir().expect("failed to get root directory");
```

Comment:
A safer way of getting this is to use the `CARGO_MANIFEST_DIR` env var, like this:
```rust
concat!(env!("CARGO_MANIFEST_DIR"), "examples/vulkan/shaders/hotload_vert.glsl");
```

I think this should give you a `&'static str` containing the full absolute path. This is a little more robust in the case that the user doesn't happen to be `cd`d into the nannou directory when they're running the example.